--- ../src-base/minecraft/net/minecraft/village/MerchantRecipe.java
+++ ../src-work/minecraft/net/minecraft/village/MerchantRecipe.java
@@ -60,7 +60,7 @@
 
     public boolean func_77393_a(MerchantRecipe p_77393_1_)
     {
-        return this.field_77403_a.func_77973_b() == p_77393_1_.field_77403_a.func_77973_b() && this.field_77402_c.func_77973_b() == p_77393_1_.field_77402_c.func_77973_b() ? this.field_77401_b == null && p_77393_1_.field_77401_b == null || this.field_77401_b != null && p_77393_1_.field_77401_b != null && this.field_77401_b.func_77973_b() == p_77393_1_.field_77401_b.func_77973_b() : false;
+        return net.minecraftforge.common.ForgeHooks.areTradesEqual(this, p_77393_1_);
     }
 
     public boolean func_77391_b(MerchantRecipe p_77391_1_)
