--- ../src-base/minecraft/net/minecraft/village/MerchantRecipeList.java
+++ ../src-work/minecraft/net/minecraft/village/MerchantRecipeList.java
@@ -22,10 +22,11 @@
 
     public MerchantRecipe func_77203_a(ItemStack p_77203_1_, ItemStack p_77203_2_, int p_77203_3_)
     {
+        MerchantRecipe target = new MerchantRecipe(p_77203_1_, p_77203_2_, null);
         if (p_77203_3_ > 0 && p_77203_3_ < this.size())
         {
             MerchantRecipe merchantrecipe1 = (MerchantRecipe)this.get(p_77203_3_);
-            return p_77203_1_.func_77973_b() == merchantrecipe1.func_77394_a().func_77973_b() && (p_77203_2_ == null && !merchantrecipe1.func_77398_c() || merchantrecipe1.func_77398_c() && p_77203_2_ != null && merchantrecipe1.func_77396_b().func_77973_b() == p_77203_2_.func_77973_b()) && p_77203_1_.field_77994_a >= merchantrecipe1.func_77394_a().field_77994_a && (!merchantrecipe1.func_77398_c() || p_77203_2_.field_77994_a >= merchantrecipe1.func_77396_b().field_77994_a) ? merchantrecipe1 : null;
+            return merchantrecipe1.func_77391_b(target) ? merchantrecipe1 : null;
         }
         else
         {
@@ -33,7 +34,7 @@
             {
                 MerchantRecipe merchantrecipe = (MerchantRecipe)this.get(j);
 
-                if (p_77203_1_.func_77973_b() == merchantrecipe.func_77394_a().func_77973_b() && p_77203_1_.field_77994_a >= merchantrecipe.func_77394_a().field_77994_a && (!merchantrecipe.func_77398_c() && p_77203_2_ == null || merchantrecipe.func_77398_c() && p_77203_2_ != null && merchantrecipe.func_77396_b().func_77973_b() == p_77203_2_.func_77973_b() && p_77203_2_.field_77994_a >= merchantrecipe.func_77396_b().field_77994_a))
+                if (merchantrecipe.func_77391_b(target))
                 {
                     return merchantrecipe;
                 }
