--- ../src-base/minecraft/net/minecraft/entity/passive/EntityVillager.java
+++ ../src-work/minecraft/net/minecraft/entity/passive/EntityVillager.java
@@ -180,7 +180,7 @@
         ItemStack itemstack = p_70085_1_.field_71071_by.func_70448_g();
         boolean flag = itemstack != null && itemstack.func_77973_b() == Items.field_151063_bx;
 
-        if (!flag && this.func_70089_S() && !this.func_70940_q() && !this.func_70631_g_())
+        if (!flag && this.func_70089_S() && !this.func_70940_q() && !this.func_70631_g_() && !p_70085_1_.func_70093_af())
         {
             if (!this.field_70170_p.field_72995_K)
             {
@@ -566,7 +566,7 @@
 
     private static ItemStack func_146088_a(Item p_146088_0_, Random p_146088_1_)
     {
-        return new ItemStack(p_146088_0_, func_146092_b(p_146088_0_, p_146088_1_), 0);
+        return new ItemStack(p_146088_0_, func_146092_b(p_146088_0_, p_146088_1_), net.minecraftforge.oredict.OreDictionary.WILDCARD_VALUE);
     }
 
     private static int func_146092_b(Item p_146092_0_, Random p_146092_1_)
